background/redux-slices/assets.ts

```diff
@@ -518,14 +519,19 @@ export const checkTokenContractDetails = createBackgroundAsyncThunk(
   }
 )
 
-transactionConstructionSliceEmitter.on("signedTransactionResult", (tx) => { 
+transactionConstructionSliceEmitter.on("signedTransactionResult", async (tx) => { 
   console.log("transaction signed", tx)
   const provider = globalThis.main.chainService.providerForNetworkOrThrow(tx.network)
-  console.log(provider instanceof JsonRpcProvider)
-  provider.sendTransaction(serializeSigned(tx)).then(res => {
+  try {
+    let res = await provider.sendTransaction(serializeSigned(tx))
     console.log(res)
     globalThis.main.chainService.saveTransaction(tx, "local")
-  })
+  } catch (error: any) {
+    if(error.toString().includes("insufficient funds")) {
+      globalThis.main.store.dispatch(setSnackbarMessage("Insufficient funds for gas * price + value"))
```

Comment:
Is there a mechanism to determine the max sendable amount at the current gas price? e.g. if someone tries to sweep their funds to another wallet

Reply:
Not exactly, because the amount spent on gas depends on the base fee which is variable between blocks

Reply:
Given a base fee, miner tip and gas limit we could calculate the maximum sendable amount